Makefile

```diff
@@ -38,4 +40,31 @@ bootstrap:
 		go get -u $$tool; \
 	done
 
+# the following targets are run in CircleCI as part of the build/test jobs
+
+# build the build image
+build-cross-image:
+	docker build -t cross-image:latest .
+
+# use the pre-built image (with dependencies set-up) to build the binary
+build-in-container: build-cross-image
+	docker run --rm \
+	-v /var/run/docker.sock:/var/run/docker.sock \
+	-v $(CURDIR)/pkg:/go/src/github.com/hashicorp/vault-plugin-database-oracle/pkg \
+	cross-image:latest
```

Comment:
Minor nit: it might be nice to specify the command `make bin` here so it's more obvious what is happening compared with the `test-in-container` target below. You could then (optionally) remove the CMD from the Dockerfile, or change it to something like `CMD echo "Please specify a command, e.g. 'make bin' or 'make test-ci'"`

Reply:
Thanks Sam!